### PR TITLE
Updating Configuration for Services

### PR DESCRIPTION
**Pull request by blankdots**
### Describe the pull request:

- [x] Bug fix
- [x] Functional change

### Pull request long description:
Changes in core services to be able to communicate with one another, by updating environment variables, configuration and other necessary parameters.

Test PR will follow, so that we can do an end to end test.

### Changes made:
1. Adding option for dataedge to use a custom public key for validating Tokens
2. DataEdge service configuration update
3. Keyserver Service configuration update
4. RES - reencryption service configuration update
5. CentraEGA message broker config now matches the one at CRG and https://github.com/EGA-archive/LocalEGA/blob/master/deploy/bootstrap/run.sh#L599-L619
6. Corrected some federation policy patterns to match: https://github.com/EGA-archive/LocalEGA/blob/master/deploy/images/mq/entrypoint.sh#L70-L81
7. Documentation update

Note that Verify, RES, Keyserver need to use the same key, password and shared password - otherwise the full cycle will not be achieved.

### Related issues:
Related to this https://github.com/EGA-archive/LocalEGA/issues/7

### Additional information:
Images used are detailed here: https://github.com/EGA-archive/LocalEGA/issues/7#issuecomment-463077056

### Documentation change:
Update parameters.

### Mentions:
@jbygdell The volumes for insert the key, shared password and actual password have been giving me trouble thus I made in separate directories instead of having a root folder as volume and some subfolders as other volumes

### CODE REVIEW

**Review by jbygdell — APPROVED** [review]
I see a couple of strategic changes in the future to simplify the configs.

File: ega-charts/localega/templates/dataedge-deploy.yaml [hunk 1]
```diff
@@ -30,12 +30,25 @@ spec:
         env:
         - name: SPRING_PROFILES_ACTIVE
           value: no-oss
+        - name: FILEDATABASE_SERVERS
+          value: {{ if .Values.config.filedatabase_host }}{{ .Values.config.filedatabase_host }}:{{ .Values.config.filedatabase_port }}{{ else }}{{ template "localega.fullname" . }}-filedatabase:{{ .Values.filedatabase.servicePort }}{{ end }}
+        - name: KEYS_SERVERS
+          value: {{ if .Values.config.keyserver_host }}{{ .Values.config.keyserver_host }}:{{ .Values.config.keyserver_port }}{{ else }}{{ template "localega.fullname" . }}-keys:{{ .Values.keys.servicePort }}{{ end }}
+        - name: RES_SERVERS
+          value: {{ if .Values.config.res_host }}{{ .Values.config.res_host }}:{{ .Values.config.res_port }}{{ else }}{{ template "localega.fullname" . }}-res:{{ .Values.res.servicePort }}{{ end }}
```
Comment (jonandernovella) [inline ega-charts/localega/templates/dataedge-deploy.yaml]: Might be good to add `quote` to the new ENV vars too

**Review by jonandernovella — CHANGES_REQUESTED** [review]
The refactoring of the variables LGTM. Good to add quoting to the newly added ENV vars.

File: ega-charts/localega/templates/dataedge-deploy.yaml (hunk 1 above)
Comment (blankdots) [reply ega-charts/localega/templates/dataedge-deploy.yaml]: Done.

**Review by jonandernovella — APPROVED** [review]
Good!

**Pull request merged**